bindings/java/src/executor.rs

```diff
@@ -128,25 +128,27 @@ pub(crate) fn make_tokio_executor(env: &mut JNIEnv, cores: usize) -> Result<Exec
     Ok(Executor::Tokio(executor))
 }
 
-/// # Safety
+/// # Panic
 ///
-/// This function could be only when the lib is loaded.
-pub(crate) unsafe fn executor_or_default<'a>(
+/// Crash if the executor is disposed.
+#[inline]
+pub(crate) fn executor_or_default<'a>(
     env: &mut JNIEnv<'a>,
     executor: *const Executor,
-) -> &'a Executor {
-    if executor.is_null() {
-        default_executor(env)
-    } else {
-        &*executor
+) -> Result<&'a Executor> {
+    unsafe {
+        if executor.is_null() {
+            default_executor(env)
+        } else {
+            // SAFETY: executor must be valid
+            Ok(&*executor)
```

Comment:
No way to catch deref invalid pointer "error". Even `catch_unwind` cannot catch it.

Reply:
I'm confused: if `executor.is_null()` has been checked, how can dereferencing still occur here?

Reply:
If the Java AsyncExecutor is disposed but still used.

Something like used a moved structure in C++ 😇

Reply:
Still don't understand, but I trust you.